packages/web/remix/index.ts

```diff
@@ -0,0 +1,384 @@
+        this.logger?.error?.(
+          '@atamaco/remix: Could not run one or more actions',
+          error,
+        );
+
+        return {
+          status: 500,
+        } as const;
+      }
+
+      return {
+        status: 200,
+        data: result,
+      } as const;
+    }
+
+    return {
+      status: 200,
+      data: result,
+    } as const;
+  }
+
+  /**
+   * Attempts to load a path from a list of paths. As soon as one path is found,
+   * it will return the data. If it cannot find a path it will throw a 404.
+   * Only throws a 500 if the fetcher throws a 500. This may happen at any of
+   * the paths in the list.
+   */
+  async loadPaths<T>(
+    args: DataFunctionArgs,
+    pathOrPaths: Array<Request | string>,
+  ) {
+    this.logger?.info?.(`@atamaco/remix: Loading paths`);
+
+    // eslint-disable-next-line no-restricted-syntax
+    for await (const urlOrRequest of pathOrPaths) {
+      try {
+        const path = resolveUrlOrRequestToPath(urlOrRequest);
+        this.logger?.debug?.(`@atamaco/remix: Attempting to load path ${path}`);
+
+        // If `loadPath` does not throw then we return the data
+        // eslint-disable-next-line no-underscore-dangle
+        const { status, data } = await this._loadPath<T>(path, args);
+
+        if (status === 200) {
+          this.logger?.debug?.(`@atamaco/remix: Loaded path ${path}`);
+          // If we get a 200 then we return the data
+          return data;
+        }
+        // Otherwise we continue the for loop
+        this.logger?.debug?.(
+          `@atamaco/remix: Could not find data for path ${path}. Continuing with next path.`,
+        );
+      } catch (error) {
+        // If a promise rejects with anything other than a 404 then we throw a
+        // 500 because that means there's an issue with the Delivery API.
+        // Reject early rather than bombarding the API with requests.
+        // eslint-disable-next-line @typescript-eslint/no-throw-literal
+        throw new Response('', { status: 500 });
+      }
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-throw-literal
+    throw new Response('', { status: 404 });
+  }
+
+  async loader<T>(args: DataFunctionArgs, pathOverride?: string) {
+    if (!pathOverride) {
+      return this.loadPaths<T>(args, [args.request]);
+    }
+
+    return this.loadPaths<T>(args, [pathOverride]);
+  }
+
+  async action<T>(args: DataFunctionArgs, pathOverride?: string) {
+    this.logger?.info?.(`@atamaco/remix: Running action`);
+    if (!this.componentTypeActions?.write) {
+      throw new Error('No write actions defined');
+    }
+
+    let path: string;
+    if (pathOverride) {
+      path = pathOverride;
+    } else {
+      const url = new URL(args.request.url);
+      path = url.pathname;
+    }
+
+    const body = await args.request.formData();
+    const {
+      actionKey: omittedActionKey,
+      componentType: omittedComponentType,
+      ...formData
+    } = Object.fromEntries(body.entries());
```

Comment:
This is interesting! I don't know that I've ever thought about this, it took me a second (despite your clear names :)) to realize you're just trying to omit the actionKey / componentType from formData.

Reply:
It's my very own lodash.omit 😄 Open to other suggestions if there's something better!